aspnet/testing/unit-testing.rst

```diff
@@ -2,28 +2,131 @@ Unit Testing
 ============
 By `Steve Smith`_
 
-This content is in progress.
+ASP.NET 5 has been designed with testability in mind, so that creating unit tests for your applications is easier than ever before. This article briefly introduces unit tests (and how they differ from other kinds of tests) and demonstrates how to add a test project to your solution and then run unit tests using either the command line or Visual Studio.
 
 In this article:
-  - `Unit Testing`_
+  - `Getting Started with Testing`_
   - `Creating Test Projects`_
   - `Running Tests`_
 
+`Download sample from GitHub <https://github.com/aspnet/docs/tree/1.0.0-beta8/aspnet/testing/unit-testing/sample>`_. 
 
-`Download sample from GitHub <https://github.com/aspnet/docs/tree/master/aspnet/testing/unit-testing/sample>`_. 
+Getting Started with Testing
+----------------------------
+Having a suite of automated tests is one of the best ways to ensure a software application does what its authors intended it to do. There are many different kinds of tests for software applications, including :doc:`integration tests <integration-testing>`, web tests, load tests, and many others. At the lowest level are unit tests, which test individual software components or methods. Unit tests should only test code within the developer's control, and should not test infrastructure concerns, like databases, file systems, or network resources. Unit tests may be written using `Test Driven Development(TDD) <http://deviq.com/test-driven-development/>`_, or they can be added to existing code to confirm its correctness. In either case, they should be small, well-named, and fast, since ideally you will want to be able to run hundreds of them before pushing your changes into the project's shared code repository.
 
-Unit Testing
-------------
-About unit testing
+.. note:: A `unit test naming convention <http://ardalis.com/unit-test-naming-convention>`_ I've found to be very effective is to name the class and test such that it forms a sentence, e.g. ``PricingCalculatorShould.ReturnZeroGivenQuantityZero()``. This makes for very readable test results, and tends to be far more helpful than naming test classes "FooTests" or "BarTests".
```

Comment:
Here are our naming guidelines: https://github.com/aspnet/Home/wiki/Engineering-guidelines#unit-test-method-naming We shouldn't make recommendations in the docs beyond what we use ourselves.

The tone is also out of place for docs, there shouldn't be any personal "I" references.